packages/common/src/items/create-content.ts

```diff
@@ -0,0 +1,52 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import { ICreateItemResponse, setItemAccess } from "@esri/arcgis-rest-portal";
+import { IItemAdd } from "@esri/arcgis-rest-types";
+import { createContentWithFile } from "./create-content-with-file";
+import { createContentWithUrl } from "./create-content-with-url";
+import { _waitForItemReady } from "./_internal/_wait-for-item-ready";
+
+/**
+ * Creates an item in online from either url or file.
+ * Once created we wait for the item to be ready (or throw an error if creation failed)
+ * If access is not private then we make a call to update that.
+ *
+ * @export
+ * @param {IItemAdd} item Item to be uploaded into online
+ * @param {IUserRequestOptions} requestOptions
+ * @return {*}  {Promise<string>} AGO item id
+ */
+export async function createContent(
```

Comment:
Similar to above. We're resolving the result from a call to RESTJS `createItem` and we're in the `items` namespace, , this may be better named `createItem`.

Reply:
As discussed went with `createItemFromUrlOrFile`